### PR TITLE
Add get_gcc_version()

### PR DESCRIPTION
**Pull request by wpoely86**
@boegel please review

### CODE REVIEW

File: easybuild/tools/systemtools.py [hunk 1]
```diff
@@ -410,6 +410,26 @@ def get_tool_version(tool, version_option='--version'):
         return '; '.join(out.split('\n'))
 
 
+def get_gcc_version():
+    """
+    Process `gcc --version` and return the GCC version.
+    """
+    out, ec = run_cmd('gcc --version', simple=False, log_ok=False, force_in_dry_run=True)
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: add `verbose=False` (we don't want to see this listed in `-x` output)?

File: easybuild/tools/systemtools.py [hunk 2]
```diff
@@ -410,6 +410,26 @@ def get_tool_version(tool, version_option='--version'):
...
+    if ec:
+        _log.warning("Failed to determine the version of GCC: %s" % out)
+        return UNKNOWN
+
+    find_version = re.search("^gcc\s+\([^)]+\)\s+(?P<version>[^\s]+)\s+", out)
+    if find_version:
+        return find_version.group('version')
+    else:
+        # Apple likes to install clang but call it gcc. It's mess, use a proper OS.
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: haha, drop the last part of the comment please :)

File: easybuild/tools/systemtools.py [hunk 3]
```diff
@@ -410,6 +410,26 @@ def get_tool_version(tool, version_option='--version'):
...
+        if get_os_type() == DARWIN:
+            return None
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: add a `log.warning`?

File: easybuild/tools/systemtools.py [hunk 4]
```diff
@@ -410,6 +410,26 @@ def get_tool_version(tool, version_option='--version'):
...
+    Process `gcc --version` and return the GCC version.
+    """
+    out, ec = run_cmd('gcc --version', simple=False, log_ok=False, force_in_dry_run=True)
+    if ec:
+        _log.warning("Failed to determine the version of GCC: %s" % out)
+        return UNKNOWN
+
+    find_version = re.search("^gcc\s+\([^)]+\)\s+(?P<version>[^\s]+)\s+", out)
+    if find_version:
+        return find_version.group('version')
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: don't return inline (use `res = find_version_group('version')`, and `return res` at the end), add a log debug mentioning both `out` and the extracted version

File: test/framework/systemtools.py [hunk 5]
```diff
@@ -380,6 +382,17 @@ def test_os_version(self):
...
+        """Test getting gcc version."""
+        gcc_version = get_gcc_version()
+        self.assertTrue(isinstance(gcc_version, basestring) or gcc_version == UNKNOWN)
+
+    def test_gcc_version_linux(self):
+        """Test getting gcc version (mocked for Linux)."""
+        st.get_os_type = lambda: st.LINUX
+        st.run_cmd = mocked_run_cmd
+        self.assertEqual(get_gcc_version(), '5.1.1')
+
```
Comment (boegel) [inline test/framework/systemtools.py]: please add a test for Darwin too, to make sure `get_gcc_version` 'works' there too?

**Comment by wpoely86** [comment]
@boegel done

File: easybuild/tools/systemtools.py [hunk 6]
```diff
@@ -410,6 +410,31 @@ def get_tool_version(tool, version_option='--version'):
...
+    out, ec = run_cmd('gcc --version', simple=False, log_ok=False, force_in_dry_run=True, verbose=False)
+    res = None
+    if ec:
+        _log.warning("Failed to determine the version of GCC: %s" % out)
+        res = UNKNOWN
+
+    find_version = re.search("^gcc\s+\([^)]+\)\s+(?P<version>[^\s]+)\s+", out)
+    if find_version:
+        res = find_version.group('version')
+        _log.debug("Found GCC version: %s from %s" % (res, out))
```
Comment (boegel) [inline easybuild/tools/systemtools.py]:
```python
_log.debug("Found GCC version: %s from %s", res, out)
```

File: easybuild/tools/systemtools.py [hunk 7]
```diff
@@ -410,6 +410,31 @@ def get_tool_version(tool, version_option='--version'):
...
+    else:
+        # Apple likes to install clang but call it gcc. <insert rant about Apple>
+        if get_os_type() == DARWIN:
+            _log.warning("On recent version of Mac OS, gcc is actually clang")
+            res = None
+        else:
+            raise EasyBuildError("Failed to determine the GCC version from: %s" % out)
```
Comment (boegel) [inline easybuild/tools/systemtools.py]:
```python
raise EasyBuildError("Failed to determine the GCC version from: %s", out)
```

File: easybuild/tools/systemtools.py [hunk 8]
```diff
@@ -410,6 +410,31 @@ def get_tool_version(tool, version_option='--version'):
...
+    """
+    Process `gcc --version` and return the GCC version.
+    """
+    out, ec = run_cmd('gcc --version', simple=False, log_ok=False, force_in_dry_run=True, verbose=False)
+    res = None
+    if ec:
+        _log.warning("Failed to determine the version of GCC: %s" % out)
+        res = UNKNOWN
+
+    find_version = re.search("^gcc\s+\([^)]+\)\s+(?P<version>[^\s]+)\s+", out)
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: add example of `gcc --version` output as a comment above (to help debug the regex if it's ever needed)

File: easybuild/tools/systemtools.py [hunk 9]
```diff
@@ -410,6 +410,31 @@ def get_tool_version(tool, version_option='--version'):
...
+    if find_version:
+        res = find_version.group('version')
+        _log.debug("Found GCC version: %s from %s" % (res, out))
+    else:
+        # Apple likes to install clang but call it gcc. <insert rant about Apple>
+        if get_os_type() == DARWIN:
+            _log.warning("On recent version of Mac OS, gcc is actually clang")
```
Comment (boegel) [inline easybuild/tools/systemtools.py]: add: `, returning None as GCC version`

File: test/framework/systemtools.py [hunk 10]
```diff
@@ -159,6 +160,7 @@ def mocked_run_cmd(cmd, **kwargs):
         "sysctl -n machdep.cpu.brand_string": "Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz",
         "sysctl -n machdep.cpu.vendor": 'GenuineIntel',
         "ulimit -u": '40',
+        "gcc --version": "gcc (GCC) 5.1.1 20150618 (Red Hat 5.1.1-4)",
```
Comment (boegel) [inline test/framework/systemtools.py]: sort alphabetically please

**Comment by boegel** [comment]
Thanks @wpoely86!

**Pull request merged**